server/datastore/mysql/hosts.go

```diff
@@ -492,8 +508,9 @@ func (ds *Datastore) applyHostFilters(opt fleet.HostListOptions, sql string, fil
 		LEFT JOIN teams t ON (h.team_id = t.id)
 		%s
 		%s
+		%s
 		WHERE TRUE AND %s AND %s
-    `, policyMembershipJoin, failingPoliciesJoin, ds.whereFilterHostsByTeams(filter, "h"), softwareFilter,
+    `, deviceMappingJoin, policyMembershipJoin, failingPoliciesJoin, ds.whereFilterHostsByTeams(filter, "h"), softwareFilter,
```

Comment:
@chiiph, I believe the device mapping info is now expected by the frontend in all cases. So unlike the `policyMembershipJoin`, I'm not aware of any frontend uses that wouldn't need this included. In practice, I think that would mean adding this new query param to all frontend requests to this endpoint. In that context, do you think we should distinguish between this and and the other cases like policies or do you think the `policyMembershipJoin` pattern is still the way to go? If the latter, what query param would you suggest?

Reply:
As for the query param... hm, `device_mapping=true`? or `include_emails=1`?

Reply:
That makes sense and seems easy enough to ask the frontend to add a new parameter to all requests. I wasn't sure which API consumers we wanted to prioritize in the design. But if we're expecting non-frontend consumers increasingly to hit this API, I definitely agree this is a good optimization. And probably a good pattern for us to build into all our our API extensions going forward. 

cc @lukeheath for his thoughts on the interface

Reply:
Sounds good. No problem for us to add a param to all hosts requests. If that param results in a slower response, agreed that we shouldn't make it default so that API users can't choose whether they want it or not. Just make sure to add it to the api docs. Thanks!